### PR TITLE
MircoRosTransport: Add C++ methods

### PR DESCRIPTION
**Pull request by nhjschulz**
Add c++ transport handler and  make the C-functions entry points only wrappers for the new c++ ones.

This removes the need for a "handcrafted" this pointer inside the C-methods for accessing Class elements.

### CODE REVIEW

File: lib/APPTurtle/src/CustomRosTransport.cpp [hunk 1]
```diff
@@ -63,81 +75,111 @@
...
+{
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->read(buffer, size, timeout, errorCode) : 0U;
+}
+
+/******************************************************************************
+ * Local Functions
+ *****************************************************************************/
+
+static CustomRosTransport* toThis(const uxrCustomTransport* transport)
```
Comment (BlueAndi) [inline lib/APPTurtle/src/CustomRosTransport.cpp:105]: Add "inline" to the local function.

File: lib/APPTurtle/src/CustomRosTransport.cpp [hunk 2]
```diff
@@ -63,81 +75,111 @@
...
 
+size_t CustomRosTransport::read(uxrCustomTransport* transport, uint8_t* buffer, size_t size, int timeout,
+                                uint8_t* errorCode)
+{
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->read(buffer, size, timeout, errorCode) : 0U;
+}
+
+/******************************************************************************
+ * Local Functions
```
Comment (BlueAndi) [inline lib/APPTurtle/src/CustomRosTransport.cpp:102]: Local functions are ususally behind the method sections.

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/APPTurtle/src/CustomRosTransport.cpp (hunk 1 above)
Comment (nhjschulz) [reply lib/APPTurtle/src/CustomRosTransport.cpp:105]: done

Comment (BlueAndi) [reply lib/APPTurtle/src/CustomRosTransport.cpp:105]: push is missing.

File: lib/APPTurtle/src/CustomRosTransport.cpp (hunk 2 above)
Comment (nhjschulz) [reply lib/APPTurtle/src/CustomRosTransport.cpp]: done

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**